x-pack/legacy/plugins/reporting/server/legacy.ts

```diff
@@ -4,35 +4,75 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import { Legacy } from 'kibana';
-import { PluginInitializerContext } from 'src/core/server';
+import { get } from 'lodash';
+import { take } from 'rxjs/operators';
+import { CoreSetup, PluginInitializerContext } from 'src/core/server';
+import { ConfigType, PluginsSetup } from '../../../../plugins/reporting/server';
 import { SecurityPluginSetup } from '../../../../plugins/security/server';
 import { ReportingPluginSpecOptions } from '../types';
 import { plugin } from './index';
-import { LegacySetup, ReportingStartDeps } from './types';
+import { LegacySetup, ReportingConfig, ReportingStartDeps } from './types';
 
 const buildLegacyDependencies = (
+  coreSetup: CoreSetup,
   server: Legacy.Server,
   reportingPlugin: ReportingPluginSpecOptions
-): LegacySetup => ({
-  config: server.config,
-  info: server.info,
-  route: server.route.bind(server),
-  plugins: {
-    elasticsearch: server.plugins.elasticsearch,
-    xpack_main: server.plugins.xpack_main,
-    reporting: reportingPlugin,
-  },
-});
+): LegacySetup => {
+  return {
+    route: server.route.bind(server),
+    plugins: {
+      xpack_main: server.plugins.xpack_main,
+      reporting: reportingPlugin,
+    },
+  };
+};
+
+const buildConfig = (
+  coreSetup: CoreSetup,
+  server: Legacy.Server,
+  reportingConfig: ConfigType
+): ReportingConfig => {
+  const config = server.config();
+  const { http } = coreSetup;
+  const serverInfo = http.getServerInfo();
+
+  const kbnConfig = {
+    path: {
+      data: config.get('path.data'), // FIXME: get from the real PluginInitializerContext
```

Comment:
Does this FIXME apply once we're out of legacy?

Reply:
dq, yup, it does

Reply:
Yes: since we're in legacy, we have to mock out the `initializerContext` variable. When we're out of legacy, we will get the data path from the a real initializerContext variable instead of passing `server`